lib/Spot/Mapper.php

```
@@ -817,4 +815,91 @@ public function isEmpty($value)
     {
         return empty($value) && !is_numeric($value);
     }
+
+    public function on($entityName, $hook, $callable)
+    {
+        if (is_callable($callable)) {
+            $this->_hooks[$entityName][$hook][] = $callable;
```

This should check the inverse, and throw an `\InvalidArgumentException` on failure. Use [@expectedException](http://www.phpunit.de/manual/3.6/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.exceptions.examples.ExceptionTest.php) annotation for this in PHPUnit.